credentials/generate-revocation-set.py

```diff
@@ -81,41 +68,109 @@ def main(use_main_net_dcld, use_test_net_dcld, use_main_net_http, use_test_net_h
     rest_node_url = PRODUCTION_NODE_URL_REST if production else TEST_NODE_URL_REST
 
     if use_rest:
-        revoked_cert_list = requests.get(f"{rest_node_url}/dcl/pki/revoked-certificates").json()["revokedCertificates"]
+        revocation_point_list = requests.get(f"{rest_node_url}/dcl/pki/revocation-points").json()["PkiRevocationDistributionPoint"]
     else:
         cmdlist = ['config', 'output', 'json']
         subprocess.Popen([dcld] + cmdlist)
 
-        cmdlist = ['query', 'pki', 'all-revoked-x509-certs']
+        cmdlist = ['query', 'pki', 'all-revocation-points']
 
         cmdpipe = subprocess.Popen(use_dcld(dcld, production, cmdlist), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
 
-        revoked_cert_list = json.loads(cmdpipe.stdout.read())["revokedCertificates"]
+        revocation_point_list = json.loads(cmdpipe.stdout.read())["PkiRevocationDistributionPoint"]
 
     revocation_set = []
 
-    for revoked_cert in revoked_cert_list:
-        serialnumber_list = []
+    for revocation_point in revocation_point_list:
+        # 1. Validate Revocation Type
+        if revocation_point["revocationType"] != 1:
+
+            # 2. Parse the certificate
+            crl_signer_certificate = x509.load_pem_x509_certificate(revocation_point["crlSignerCertificate"])
+
+            vid = revocation_point["vid"]
+            pid = revocation_point["pid"]
```

Comment:
The VID or PID can be embedded in the subject as well. We need to factor the extraction into methods.